googleapiclient/discovery.py

```diff
@@ -196,21 +200,23 @@ def build(serviceName,
   if http is None:
     http = httplib2.Http()
 
-  requested_url = uritemplate.expand(discoveryServiceUrl, params)
-
-  try:
-    content = _retrieve_discovery_doc(requested_url, http, cache_discovery,
-                                      cache)
-  except HttpError as e:
-    if e.resp.status == http_client.NOT_FOUND:
-      raise UnknownApiNameOrVersion("name: %s  version: %s" % (serviceName,
-                                                               version))
-    else:
-      raise e
+  for discovery_url in [discoveryServiceUrl, V2_DISCOVERY_URI]:
```

Comment:
The shape of this code may change in response to my question about the default value of the `discoveryServiceUrl` parameter, but with respect to the current draft: why try the old thing and then fall back to the new thing? Why not try the new thing first and then fall back to the old thing if necessary?